### DISCUSSION TITLE
Generate a mindmap using the Excalidraw Plugin and Templater

### DISCUSSION THREAD

**Discussion by zsviczian** [SilentVoid13/Templater#202] [Templates Showcase]
Here's the [Gist](https://gist.github.com/zsviczian/a3af75433320d09b21e481478f2d8b3a)

You can find full documentation of ExcalidrawAutomate [here](https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/master/AutomateHowTo.md).

This generates a mindmap that you can further edit from a tabulated outline.
![image](https://user-images.githubusercontent.com/14358394/117496599-8dce2b00-af77-11eb-85ba-5d31054c13bd.png)

**Comment by OliverBalfour** [comment]
Great template, thanks!

I've just found that it doesn't work if you use `*` instead of `-` for bullet points. I've attached a simple fix below if anyone else has this issue.

Fix: replace the `getLineProps` section with this one
```js
function getLineProps (i) {
  props = editor.getLine(i).match(/^(\t*)(-|\*)\s+(.*)/);
  return [props[1].length+1, props[3]];
}
```

**Reply by hehelaughs** [reply]
how to generate a mindmap according to “#” instead of "-"?

**Reply by hehelaughs** [reply]
(-|\#),it doesn't work

**Reply by hehelaughs** [reply]
/^(\#{1,3})\s(.*)/
why not?

**Reply by zsviczian** [reply]
I suggest you refine your regular expression using https://regexr.com/
You can paste a sample document there, and see your regular expression in action. regexr.com is a must for any regexp development

**Reply by zsviczian** [reply]
btw your expression has fewer capture groups than the example above, thus if your expression is `/^(#{1,3})\s(.*)/` then `props[3]` will be `undefined`

**Comment by jroachgit** [comment]
Any thoughts on how to expand this text > template > code > draw,  to develop mermaid like workflows, like BPMN or UML?

**Comment by tyf2018** [comment]
Thanks so much for sharing!

I encountered a problem. Using the MindmapInput.md provided by you, the conversion was successful, but the md file I created by myself always reported an error, and the console displayed: Cannot read property '1' of null.

This is the md file I created by myself. Can you give me some advice? Thanks!

[test.md](https://github.com/SilentVoid13/Templater/files/7282685/test.md)

**Reply by garciaErick** [reply]
same

**Comment by zsviczian** [comment]
The sample script does not have any error handling and it fails if it finds
empty lines at the end of your file. Simply delete the two trailing empty
lines and the file will work. You can play with the code to handle this
situation.

**Reply by tyf2018** [reply]
It was indeed caused by the  two blank lines, and now it works normally, thank you.
